openslide/_version.py

```diff
@@ -23,3 +23,12 @@
 obtained from openslide.__version__."""
 
 __version__ = '1.1.2'
```

Comment:
Hmm, I should have pulled this line.

We can't pull the version into pyproject.toml, so we have to flip the script and pull the version from the installed package metadata. With poetry's development model we always work with an installed version in a virtualenv, so the version "should" always exist.

Reply:
It does something like pip install -e, so you can still edit in the source directory. But you normally run from the virtualenv with `poetry run <cmd>`. Because openslide has no cli I normally use `poetry run python` and then `import openslide` and go from there. I'm sure there are many alternate ways, like use `poetry shell` to get a shell, which is similar to `. venv/bin/activate` with virtualenvs.